core/src/banking_stage.rs

```diff
@@ -679,22 +645,44 @@ impl BankingStage {
                     // duplicate signature, etc.)
                     //
                     // Note: This assumes that every packet deserializes into one transaction!
-                    consumed_buffered_packets_count += original_unprocessed_indexes
-                        .len()
-                        .saturating_sub(retryable_transaction_indexes.len());
+                    consumed_buffered_packets_count +=
+                        packets_to_process_len.saturating_sub(retryable_transaction_indexes.len());
 
                     // Out of the buffered packets just retried, collect any still unprocessed
                     // transactions in this batch for forwarding
                     rebuffered_packet_count += retryable_transaction_indexes.len();
-                    let has_more_unprocessed_transactions = deserialized_packet_batch
-                        .update_buffered_packets_with_new_unprocessed(
-                            &original_unprocessed_indexes,
-                            &retryable_transaction_indexes,
-                        );
                     if let Some(test_fn) = &test_fn {
                         test_fn();
                     }
-                    has_more_unprocessed_transactions
+
+                    slot_metrics_tracker.increment_retryable_packets_count(
+                        retryable_transaction_indexes.len() as u64,
+                    );
+
+                    let result = retryable_transaction_indexes
+                        .iter()
+                        .map(|i| packets_to_process[*i].clone())
```

Comment:
this just cloning the `PacketPriorityQueueEntry`, which is two `u64` and a `Rc`, should be cheap.